### PR TITLE
fix(jtk): handle array of component/version in --field formatter

### LINKED ISSUE #227: jtk: cannot parse the components field with JSON array list items
## Description
The jtk issues create command fails to create issues when using the `--field` flag to set components, regardless of the format used (object with id, name, or value properties, or plain string).

All attempts result in the error `The list contains an invalid value` even when using valid component IDs that are confirmed to exist in the project via `jtk issues field-options components`.
                                                                                                                                                                                                 
  This prevents users from assigning components when creating issues through the CLI, forcing them to manually add components after issue creation.

## Step to reproduce
- Create a list of components
- Try to create an issue with the components list using the command below
- Check the error message below is returning in all the attempts

```
jtk issues create
  --project MTA
  --type Task
  --summary "Waddle detection improvement"
  --description "Penguins are birds that cannot fly, but they are excellent swimmers."
  --field 'components=[{"id":"10201"}]'
```
More options tried with the same error (extracted from here)
https://developer.atlassian.com/server/jira/platform/jira-rest-api-example-create-issue-7897248/
```
  --field 'components=[{"id":10201}]'
  --field 'components=[{"value":"Service 2"}]'
  --field 'components=[{"name":"Service 2"}]'
  --field "components=Service 2"
  --components "Service 2"
```

## Expected behavior
The issue should be created successfully

## Actual behavior
The issue is not created successfully and return the error message:

```
creating issue: bad request: components: The list contains an invalid value
```

## Additional information

Components array list used:

```
% jtk fields list

ID               NAME               TYPE          CUSTOM
components       Components         array         no
...
```

Components values:

```
% jtk issues field-options components --issue MTA-100

Allowed values for field 'Components':
VALUE       ID
Service 1   10200
Service 2   10201
Service 3   10202
Service 4   10203
```

JTK version: 
```
jtk version 1.0.56
```

### PR DESCRIPTION
**Pull request by rianjs**
## Summary

Fixes #227, originally reported by @romiguelangel who also opened a fix in their fork: https://github.com/romiguelangel/atlassian-cli/pull/1. Pulled the conceptual change into upstream main with attribution preserved on the commit author.

`jtk issues create --field components=<id-or-name>` (and `fixVersions=...`) now works. Before this PR, every input shape — id, name, JSON object, JSON array — produced `creating issue: bad request: components: The list contains an invalid value`.

## Root cause

`FormatFieldValue` in `tools/jtk/api/fields.go`:

```go
case "array":
    if field.Schema.Items == "option" {
        return []map[string]string{{"value": trimmed}}
    }
    return []string{trimmed}   // ← components fell here, Jira rejects
```

The components field has `schema.items: component`, which fell through to the plain string-array path. Jira's create-issue endpoint expects `[{"id":"..."}]` or `[{"name":"..."}]` — not `["Frontend"]`.

## Fix

Added an `Items in {"component", "version"}` branch following the same pattern as `priority/resolution/status` at line 162: numeric input → `{id}`, otherwise → `{name}`.

## Why not group, and why not the JSON pass-through

The contributor's original PR also added `Items == "group"` and a generic `if value starts with [ or {, json.Unmarshal` escape hatch. Dropped both:

- **`group`**: Atlassian's example shows multi-group custom fields as plain string arrays (`["jira-administrators", ...]`), which the existing `[]string` fallback handles correctly. Adding `group` to the new branch would convert working calls into `[{"name":"..."}]` and risk regression. Added a regression-guard test to lock the existing behavior.
- **JSON pass-through**: bypasses schema-aware formatting for any field whose value happens to start with `[` or `{`. Surprising for textarea (ADF), options, etc. The right way to add this back later is an explicit `--field-json key=<json>` flag, opt-in by design — not prefix-sniffing inside `FormatFieldValue`.

## Test plan

- [x] Unit tests in `tools/jtk/api/fields_test.go` cover `array of component/version` by id and name + a `group` regression guard
- [x] Merge test for accumulating components across repeated `--field`
- [x] Full create-path wiring test in `tools/jtk/internal/cmd/issues/create_test.go` using httptest with both `GET /rest/api/3/field` and captured `POST /rest/api/3/issue` body — exercises parse → metadata → format → merge → POST end-to-end, and asserts the existing multi-checkbox `[{value: ...}]` shape still merges correctly
- [x] `cd tools/jtk && go test ./...` → 1748 pass
- [x] Live verification deferred to merge: `jtk issues create --project MON --type SDLC --summary "test" --field 'components=Admin Portal'`

Closes #227

### CODE REVIEW

**Comment by rianjs** [comment]
**Findings**

None.

The PR adheres to the agreed architecture: component/version only, `group` preserved on the existing `[]string` fallback with a regression guard, no generic JSON prefix parsing, and the create-path test covers metadata lookup, formatting, merge behavior, and outbound body shape. The residual numeric-as-ID heuristic is intentional and consistent with the existing formatter contract.

**Comment by rianjs** [comment]
## Test Coverage Assessment

Coverage of the `components`/`fixVersions` array fix is solid. The unit tests cover the four important branches (component-by-id, component-by-name, version-by-id, version-by-name), include a regression guard for `array of group` falling through to the plain string array path, and the `MergeFieldValues` test confirms id-form component arrays accumulate correctly. The wiring test in `create_test.go` is the right shape — it captures the actual POST body and asserts on the marshalled JSON, which is what would have caught the original bug end-to-end.

### Minor

- **Missing edge case: leading/trailing whitespace + numeric.** `FormatFieldValue` already trims, but there's no test that `" 10201 "` still hits the id path. Not load-bearing, but cheap to add.
- **Missing edge case: alphanumeric-with-leading-digit name.** A value like `"12abc"` should take the `name` path (since `strconv.Atoi` rejects it). No test asserts this — worth one row in the table to lock the contract.
- **Merge test only covers id form.** Add a name-form merge case (`[{name: "A"}] + [{name: "B"}]`) and ideally a mixed id/name merge to mirror real usage.

### Nit

- The wiring test asserts `components[0].name == "Frontend"` but doesn't assert the absence of an `id` key on that map. A future bug that emits `{"name": "Frontend", "id": ""}` would slip past. `testutil.Len(t, components[0], 1)` would tighten it.
- Empty-string value behavior for `components=` is untested; trim makes it `""`, which would currently produce `[{"name": ""}]` — probably should be rejected upstream, but worth confirming intent.

### Verdict

Adequate. Tests are not vacuous and exercise both the unit and the integration seam. The minor gaps are about hardening, not correctness of the fix.

**Comment by rianjs** [comment]
Addressed TDD findings in 54aa7f1:
- Trimmed-numeric (` 10201 `) covered — confirms whitespace-trim path
- Leading-digit name (`12abc`) covered — confirms strconv.Atoi rejection routes to name
- `MergeFieldValues` now covers name-form and mixed id/name accumulation
- Wiring test locks the components[0] shape: exactly {"name": "Frontend"}, no leaked `id` key

Ran `go test ./api/... ./internal/cmd/issues/... -run TestFormatFieldValue|TestMergeFieldValues|TestRunCreate_FieldComponentArrayAndOptionMerge` → 46 pass.

File: tools/jtk/api/fields.go [hunk 1]
```diff
@@ -143,6 +144,12 @@ func FormatFieldValue(field *Field, value string) any {
 		if field.Schema.Items == "option" {
 			return []map[string]string{{"value": trimmed}}
 		}
+		if field.Schema.Items == "component" || field.Schema.Items == "version" {
```
Comment (monit-reviewer) [inline tools/jtk/api/fields.go:147]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): strconv.Atoi accepts negative integers (e.g., "-1"), which would be routed to the id path as {"id": "-1"}. Jira component/version IDs are always positive. Consider guarding with `n, err := strconv.Atoi(trimmed); err == nil && n > 0` so negatives fall through to the name path and surface a clearer error.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/issues/create_test.go [hunk 2]
```diff
@@ -631,3 +631,65 @@ func TestRunCreate_IDOnly(t *testing.T) {
...
+	testutil.RequireNoError(t, err)
+
+	opts := &root.Options{Stdout: &bytes.Buffer{}, Stderr: &bytes.Buffer{}, IDOnly: true}
+	opts.SetAPIClient(client)
+
+	err = runCreate(context.Background(), opts, "PROJ", "Task", "Test", "", "", "", []string{
+		"components=Frontend",
+		"customfield_10100=Opt1",
+		"customfield_10100=Opt2",
+	})
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/create_test.go:677]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): Unguarded type assertions (e.g., fields["components"].([]any), components[0].(map[string]any)) will panic with an opaque message on shape mismatch rather than producing a clear test failure. Prefer ok-idiom assertions or testutil helpers.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/issues/create_test.go [hunk 3]
```diff
@@ -631,3 +631,65 @@ func TestRunCreate_IDOnly(t *testing.T) {
...
+			fields := []api.Field{
+				{ID: "components", Name: "Components", Schema: api.FieldSchema{Type: "array", Items: "component"}},
+				{ID: "customfield_10100", Name: "Tags", Custom: true, Schema: api.FieldSchema{Type: "array", Items: "option"}},
+			}
+			_ = json.NewEncoder(w).Encode(fields)
+		case r.URL.Path == "/rest/api/3/issue" && r.Method == "POST":
+			capturedBody, _ = io.ReadAll(r.Body)
+			w.WriteHeader(http.StatusCreated)
+			_ = json.NewEncoder(w).Encode(api.Issue{Key: "TEST-1", ID: "10001"})
+		default:
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/create_test.go:660]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): The httptest server returns 404 for any unrecognised path. A `t.Fatalf("unexpected request: %s %s", r.Method, r.URL.Path)` default case would make future regressions easier to diagnose.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `54aa7f1`

> **Approved** with 3 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-enforcement-reviewer | 1 |
| harness-engineering:harness-self-documenting-code-reviewer | 2 |

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/api/fields.go:147`
> strconv.Atoi accepts negative integers (e.g., "-1"), which would be routed to the id path as {"id": "-1"}. Jira component/version IDs are always positive. Consider guarding with `n, err := strconv.Atoi(trimmed); err == nil && n > 0` so negatives fall through to the name path and surface a clearer error.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/issues/create_test.go:676`
> Unguarded type assertions (e.g., fields["components"].([]any), components[0].(map[string]any)) will panic with an opaque message on shape mismatch rather than producing a clear test failure. Prefer ok-idiom assertions or testutil helpers.

**💡 Suggestion** - `tools/jtk/internal/cmd/issues/create_test.go:660`
> The httptest server returns 404 for any unrecognised path. A `t.Fatalf("unexpected request: %s %s", r.Method, r.URL.Path)` default case would make future regressions easier to diagnose.

</details>

*2 info-level observations excluded. Run with `--verbose` to include.*

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 1m 16s | $0.65</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, documentation:docs-reviewer, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 1m 16s wall · 4m 10s compute (Reviewers: 48s · Synthesis: 26s) |
| Cost | $0.65 |
| Tokens | 195.1k in / 16.8k out |
| Turns | 14 |

</details>

File: tools/jtk/api/fields.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/api/fields.go:147]: Disagree on the negative-int guard. The existing `priority/resolution/status` switch (fields.go line 162) uses `strconv.Atoi` without a `n > 0` check. Adding the guard only here would create asymmetric behavior across the formatter — better as a sweep across all numeric-id branches in a follow-up if Jira actually rejects negative IDs in practice. Holding consistent with the existing pattern.

File: tools/jtk/internal/cmd/issues/create_test.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/create_test.go:677]: Fixed in fde295b — replaced unguarded type assertions with ok-idiom + `testutil.True` so shape mismatches produce a clear test failure.

File: tools/jtk/internal/cmd/issues/create_test.go (hunk 3 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/create_test.go:660]: Fixed in fde295b — default case now `t.Errorf`s on unexpected request before returning 404.
